mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/impl/MongoDBIndexesDefiner.java

```diff
@@ -0,0 +1,270 @@
+					}
+				}
+			}
+		}
+	}
+
+	void createIndexes(MongoDatabase database) {
+		for ( MongoDBIndexSpec indexSpec : indexSpecs ) {
+			createIndex( database, indexSpec );
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	private void validateIndexSpecsForUniqueColumns(Table table, IndexOptions indexOptions, Set<String> forIndexNotReferenced,
+			UniqueConstraintSchemaUpdateStrategy constraintMethod) {
+		Iterator<Column> columnIterator = table.getColumnIterator();
+		while ( columnIterator.hasNext() ) {
+			Column column = columnIterator.next();
+			if ( column.isUnique() ) {
+				String indexName = NamingHelper.INSTANCE.generateHashedConstraintName(
+						"UK_",
+						table.getNameIdentifier(),
+						Identifier.toIdentifier( column.getName() ) );
+				forIndexNotReferenced.remove( indexName );
+
+				if ( constraintMethod != UniqueConstraintSchemaUpdateStrategy.SKIP ) {
+					MongoDBIndexSpec indexSpec = new MongoDBIndexSpec( table.getName(), column.getName(), indexName,
+							getIndexOptionDocument( table, indexOptions.getOptionForIndex( indexName ) ) );
+					if ( validateIndexSpec( indexSpec ) ) {
+						addHandleDuplications( indexSpec );
+					}
+				}
+			}
+		}
+	}
+
+	private void validateIndexSpecsForUniqueKeys(Table table, IndexOptions indexOptions, Set<String> forIndexNotReferenced,
+			UniqueConstraintSchemaUpdateStrategy constraintMethod) {
+		Iterator<UniqueKey> keys = table.getUniqueKeyIterator();
+		while ( keys.hasNext() ) {
+			UniqueKey uniqueKey = keys.next();
+			forIndexNotReferenced.remove( uniqueKey.getName() );
+
+			if ( constraintMethod != UniqueConstraintSchemaUpdateStrategy.SKIP ) {
+				MongoDBIndexSpec indexSpec = new MongoDBIndexSpec( uniqueKey,
+						getIndexOptionDocument( table, indexOptions.getOptionForIndex( uniqueKey.getName() ) ) );
+				if ( validateIndexSpec( indexSpec ) ) {
+					addHandleDuplications( indexSpec );
+				}
+			}
+		}
+	}
+
+	void validateIndexSpecsForIndexes(Table table, IndexOptions indexOptions, Set<String> forIndexNotReferenced) {
+		Iterator<Index> indexes = table.getIndexIterator();
+		while ( indexes.hasNext() ) {
+			Index index = indexes.next();
+			forIndexNotReferenced.remove( index.getName() );
+			MongoDBIndexSpec indexSpec = new MongoDBIndexSpec( index,
+					getIndexOptionDocument( table, indexOptions.getOptionForIndex( index.getName() ) ) );
+			if ( validateIndexSpec( indexSpec ) ) {
+				addHandleDuplications( indexSpec );
+			}
+		}
+	}
+
+	private void addHandleDuplications(MongoDBIndexSpec indexSpec) {
+		// we can add any index defined on different collection or different key set
+		// we rely on hashcode and equals to verify conflicts
+		if ( !indexSpecs.contains( indexSpec ) ) {
+			indexSpecs.add( indexSpec );
+			return;
+		}
+
+		int existingIndexPosition = indexSpecs.indexOf( indexSpec );
+		MongoDBIndexSpec existingIndex = indexSpecs.get( existingIndexPosition );
+
+		// in case on conflict
```

Comment:
s/on/of/

Reply:
I don't like anymore the comment. I'm going to change it.